pkg/api/component_report.go

```diff
@@ -348,31 +306,27 @@ func (c *componentReportGenerator) getTestStatusFromBigQuery() (
 	[]error,
 ) {
 	errs := []error{}
-	queryString := fmt.Sprintf(`WITH latest_component_mapping AS (
-						SELECT *
-						FROM ci_analysis_us.component_mapping cm
-						WHERE created_at = (
-								SELECT MAX(created_at)
-								FROM openshift-gce-devel.ci_analysis_us.component_mapping))
-					SELECT
-						ANY_VALUE(test_name) AS test_name,
-						ANY_VALUE(testsuite) AS test_suite,
-						cm.id as test_id,
-						network,
-						upgrade,
-						arch,
-						platform,
-						flat_variants,
-						ANY_VALUE(variants) AS variants,
-						COUNT(cm.id) AS total_count,
-						SUM(success_val) AS success_count,
-						SUM(flake_count) AS flake_count,
-						ANY_VALUE(cm.component) AS component,
-						ANY_VALUE(cm.capabilities) AS capabilities,
-						ANY_VALUE(cm.jira_component) AS jira_component,
-						ANY_VALUE(cm.jira_component_id) AS jira_component_id
-					FROM (%s)
-					INNER JOIN latest_component_mapping cm ON testsuite = cm.suite AND test_name = cm.name`, dedupedJunitTable)
+	queryString := fmt.Sprintf(`
+          SELECT
+            ANY_VALUE(test_name) AS test_name,
+            ANY_VALUE(test_suite) AS test_suite,
+            test_id,
+            network,
+            upgrade,
+            arch,
+            platform,
+            flat_variants,
+            ANY_VALUE(variants) AS variants,
+            SUM(total_count) AS total_count,
+            SUM(success_count) AS success_count,
+            SUM(flake_count) AS flake_count,
+            ANY_VALUE(component) AS component,
+            ANY_VALUE(capabilities) AS capabilities,
+            ANY_VALUE(jira_component) AS jira_component,
+            ANY_VALUE(jira_component_id) AS jira_component_id
+          FROM
+            %s.junit_precomputed_by_test_and_day
+          WHERE date >= @From AND date <= @To`, c.client.Dataset)
```

Comment:
Can you clarify the use of flat_variants and variants (repeated) in the schema for this table. Just plotting for the externalized job variants work coming.

Reply:
Both have the same data, variants is the array, flat_variants is the comma-separated string

Reply:
Sorry I was wondering why both were present.

Reply:
I'm not sure, it's the existing SQL Ken wrote.  I'm guessing it's to do with bigquery limitations on working with the repeated  column (arrays)

Reply:
Actually it's coming from the junit table that Justin created, I don't know why there's both.  Probably for flexibility?